test/Snowflake_Tests/src/Snowflake_Spec.enso

```diff
@@ -482,76 +506,78 @@ snowflake_specific_spec suite_builder create_connection_fn db_name setup =
 
         group_builder.specify "round returns the correct type" <|
-            do_round data 231.2 1 . should_be_a Float
-            do_round data 231.2 0 . should_be_a Float
-            do_round data 231.2 . should_be_a Float
-            do_round data 231.2 -1 . should_be_a Float
+            do_round 231.2 1 . should_be_a Float
+            do_round 231.2 0 . should_be_a Float
+            do_round 231.2 . should_be_a Float
+            do_round 231.2 -1 . should_be_a Float
 
         group_builder.specify "round returns the correct type" <|
-            do_round data 231 1 . should_be_a Float
-            do_round data 231 0 . should_be_a Float
-            do_round data 231 . should_be_a Float
-            do_round data 231 -1 . should_be_a Float
+            do_round 231 1 . should_be_a Float
+            do_round 231 0 . should_be_a Float
+            do_round 231 . should_be_a Float
+            do_round 231 -1 . should_be_a Float
 
 add_snowflake_specs suite_builder create_connection_fn db_name =
     prefix = "[Snowflake] "
     name_counter = Ref.new 0
 
-    table_builder columns connection=(create_connection_fn Nothing) =
+    ## We prefer to keep a single connection for most tests, to avoid the overhead of initializing a new connection multiple times.
+       It is initialized lazily, so that it is actually established only if actually used. Merely listing the tests to run should not establish the connection.
+    default_connection = Lazy_Ref.new (create_connection_fn Nothing)
+    table_builder columns connection=Nothing =
         ix = name_counter.get
         name_counter . put ix+1
         name = Name_Generator.random_name "table_"+ix.to_text
 
         in_mem_table = Table.new columns
-        in_mem_table.select_into_database_table connection name primary_key=Nothing temporary=True
-
+        in_mem_table.select_into_database_table (connection.if_nothing default_connection.get) name primary_key=Nothing temporary=True
+    light_table_builder columns =
+        default_connection.get.base_connection.create_literal_table (Table.new columns) "literal_table"
     materialize = .read
 
     Common_Spec.add_specs suite_builder prefix create_connection_fn
 
-    common_selection = Common_Table_Operations.Main.Test_Selection.Config supports_case_sensitive_columns=True order_by_unicode_normalization_by_default=True allows_mixed_type_comparisons=False fixed_length_text_columns=True removes_trailing_whitespace_casting_from_char_to_varchar=True supports_decimal_type=True supported_replace_params=supported_replace_params
+    common_selection = Common_Table_Operations.Main.Test_Selection.Config supports_case_sensitive_columns=True order_by_unicode_normalization_by_default=True allows_mixed_type_comparisons=False fixed_length_text_columns=True removes_trailing_whitespace_casting_from_char_to_varchar=True supports_decimal_type=True supported_replace_params=supported_replace_params run_advanced_edge_case_tests_by_default=False
```

Comment:
Is there a particular reason not to run the advanced edge cases? Are they slow?

Reply:
Yes, exactly. We are running **thousands** of queries in the tests. In Postgres each query runs ~1ms so that's maybe acceptable but for Snowflake where queries are closer to 100ms mark - this makes the whole suite run for about an hour. They can be ran on a special nightly run that is OK if running for a longer time.